.github/workflows/push-image-on-push-to-branch.yml

```diff
@@ -0,0 +1,45 @@
+name: 'Image to GHCR on push to branch'
+
+on:
+  push:
+    branches-ignore:
+      - 'main'
+
```

Comment:
The concurrency feature has now been added on workflow level.

The if condition in the build_and_push job was changed to run only on success(). Otherwise the Job will run even when the workflow was cancelled.